internal/clients/http/sources/sources_client.go

```diff
@@ -249,49 +247,19 @@ func (c *sourcesClient) loadAppId(ctx context.Context) (string, error) {
 	return "", http.ApplicationTypeNotFoundErr
 }
 
-//nolint:exhaustive
-func filterSourceAuthentications(authentications []AuthenticationRead) (AuthenticationRead, error) {
-	for _, auth := range authentications {
-		if *auth.ResourceType == "Application" {
-			switch *auth.Authtype {
-			// Type of the authentication as stored in Sources by listing the source types or the application types
-			case "provisioning-arn",
-				"provisioning_lighthouse_subscription_id",
-				"provisioning_project_id":
-				return auth, nil
-			default:
-				continue
-			}
+func BuildQuery(keysAndValues ...string) func(ctx context.Context, req *stdhttp.Request) error {
+	return func(ctx context.Context, req *stdhttp.Request) error {
+		if len(keysAndValues)%2 != 0 {
+			return http.NotEvenErr
 		}
-	}
-	return AuthenticationRead{}, http.ApplicationReadErr
-}
-
-func (c *sourcesClient) GetSourceTypeName(ctx context.Context, sourceTypeID string) (models.ProviderType, error) {
-	logger := logger(ctx)
-	ctx, span := otel.Tracer(TraceName).Start(ctx, "GetSourceTypeName")
-	defer span.End()
-
-	// Get all the source types
-	resp, err := c.client.ListSourceTypesWithResponse(ctx, &ListSourceTypesParams{}, headers.AddSourcesIdentityHeader)
-	if err != nil {
-		return models.ProviderTypeUnknown, fmt.Errorf("cannot list source types: %w", err)
-	}
-
-	for _, st := range *resp.JSON200.Data {
-		if sourceTypeID == *st.Id {
-			logger.Trace().Msg("Found source type id from sources")
-			switch *st.Name {
-			case "amazon":
-				return models.ProviderTypeAWS, nil
-			case "google":
-				return models.ProviderTypeGCP, nil
-			case "azure":
-				return models.ProviderTypeAzure, nil
-			default:
-				return models.ProviderTypeUnknown, fmt.Errorf("provider unknown %w", clients.UnknownProviderErr)
-			}
+		queryParams := make([]string, 0)
+		for i := 0; i < len(keysAndValues); i += 2 {
+			key := url.QueryEscape(keysAndValues[i])
+			value := url.QueryEscape(keysAndValues[i+1])
+			queryParams = append(queryParams, fmt.Sprintf("%s=%s", key, value))
 		}
+
+		req.URL.RawQuery = strings.Join(queryParams, "&")
+		return nil
 	}
-	return models.ProviderTypeUnknown, fmt.Errorf("cannot find source type name for source type id %s: %w", sourceTypeID, http.SourceTypeNameNotFoundErr)
 }
```

Comment:
There is no need to return error in this case since no other function here does return one. Generally `panic` should not be used, but one of the rare cases when it is useful are static errors (e.g. this one will only happen when someone changes arguments badly).

But it’s okay to leave as is :-)